backend/app/engine/octant_rewards/matched/percentage_from_staking.py

```diff
@@ -0,0 +1,21 @@
+from app.engine.octant_rewards.matched import MatchedRewards, MatchedRewardsPayload
+from dataclasses import dataclass
+from decimal import Decimal
+
+
+@dataclass
+class PercentageMatchedRewards(MatchedRewards):
+    MATCHED_REWARDS_PERCENT: Decimal
+
+    def calculate_matched_rewards(self, payload: MatchedRewardsPayload) -> int:
+        if payload.locked_ratio < payload.ire_percent:
+            return int(
+                self.MATCHED_REWARDS_PERCENT * payload.staking_proceeds
+                + payload.patrons_rewards
+            )
+        elif payload.ire_percent <= payload.locked_ratio < payload.tr_percent:
+            return int(
+                (payload.tr_percent - payload.locked_ratio) * payload.staking_proceeds
+                + payload.patrons_rewards
+            )
+        return payload.patrons_rewards
```

Comment:
returning `patrons_rewards` when `locked_ratio` is bigger then `tr_percent` makes logical sense, but this strategy doesn't make sense as a whole when `locked_ratio` is bigger then 70%. For me throwing exception would be better in that case, but it's just a suggestion, it can stay as it is